lib/src/revive.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/constant/value.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/src/dart/constant/value.dart';
+
+import 'utils.dart';
+
+/// Returns a revivable instance of [object].
+///
+/// Optionally specify the [library] type that contains the reference.
+///
+/// Returns [null] if not revivable.
+Revivable reviveInstance(DartObject object, [LibraryElement library]) {
+  library ??= object.type.element.library;
+  var url = Uri.parse(urlOfElement(object.type.element));
+  final clazz = object?.type?.element as ClassElement;
+  for (final e in clazz.fields.where(
```

Comment:
why does this need to be a for loop if the first line is a return? Is this a typical pattern to handle the 0 or first case?

would this be more readable as

```
var matchingFields = clazz.fields.where((f) => ...);
if (matchingFields.isNotEmpty) {
return new Revivable._(source: url, accessor: matchinfFields.first.name);
}